src/Common/src/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -59,6 +59,10 @@ public static bool IsBlittableType(TypeDesc type)
         public static bool IsManagedSequentialType(TypeDesc type)
         {
             type = type.UnderlyingType;
+            if (type is MetadataType metadataType && metadataType.IsExplicitLayout)
```

Comment:
Is this doing the right thing for Auto layout?

Reply:
The corresponding logic in CoreCLR around

https://github.com/dotnet/coreclr/blob/0bdf613f09a3bf409717beb939195b39e40c6828/src/vm/class.cpp#L3635

is quite complicated. I tried to improved the condition here in 2nd commit, thankfully the JIT\Directed tests are quite sensitive to it.